app_package/src/watch.ts

```diff
@@ -0,0 +1,325 @@
+import { AbstractMesh, AnimationGroup, Bone, ISceneLoaderAsyncResult, Matrix, Observable, PBRMaterial, Scene, SceneLoader, Texture, TmpVectors, Tools, TransformNode, Vector2, Vector3 } from "@babylonjs/core";
+import { AdvancedDynamicTexture, TextBlock } from "@babylonjs/gui";
+import { IVaporwearExperienceParams } from "./iVaporwearExperienceParams";
+
+export enum WatchState {
+    Overall,
+    Clasp,
+    Face,
+    Levitate,
+    Configure
+}
+
+class WatchStateHelpers {
+    public static IsUpState(state: WatchState): boolean {
+        switch (state) {
+            case WatchState.Overall:
+            case WatchState.Clasp:
+            case WatchState.Configure:
+                return true;
+            default:
+                return false;
+        }
+    }
+}
+
+export class HotspotState {
+    constructor () {
+        this.position = new Vector2();
+        this.isVisible = false;
+    }
+
+    position: Vector2;
+    isVisible: boolean;
+}
```

Comment:
this class looks inconsistent with the rest of the code:

```suggestion
export class HotspotState {
    public position = new Vector2();
    public isVisible = false;
}
```